src/rgw/driver/rados/rgw_user.cc

```diff
@@ -1552,6 +1552,7 @@ static void rename_swift_keys(const rgw_user& user,
   user.to_str(user_id);
 
   auto modify_keys = std::move(keys);
+  keys.clear();
```

Comment:
Note that I cannot find a clear guarantee in the C++ docs that the clear() on a moved-from map does what you want it to do here.

Reply:
The specific contents or state of the keys is not guaranteed. The container might be empty, partially empty, or in some other state. We add the clear() in order to make sure that the keys is completely empty.

reference: https://en.cppreference.com/w/cpp/container/map/clear

Reply:
After discussing this with some C++ people: clear() is not explicitly allowed on a moved-from map.
Just assign an empty map to it.

Reply:
> After discussing this with some C++ people: clear() is not explicitly allowed on a moved-from map.
> Just assign an empty map to it.

i'm curious about this part. my understanding is that clear() is safe to call because it has no preconditions

Reply:
for reference, under https://en.cppreference.com/w/cpp/utility/move#Notes

> str.clear(); // OK, clear() has no preconditions

Reply:
the comment there is about `std::string`, but would make sense that `std::map::clear()` does not have preconditions either.
btw, when running "sonar" against such code, it complains on user-after-move, but not for the "clear()" call...

either way, there is probably not much to gain by using "clear" in this case